src/mapP.js

```diff
@@ -1,8 +1,30 @@
 const curry = require('ramda/src/curry')
 const map   = require('ramda/src/map')
 
-// mapP :: Functor f => (a -> Promise b) -> f a -> Promise f b
-const mapP = (f, list) =>
+const mapArrayP = (f, list) =>
   Promise.all(map(f, list))
 
+const mapObjectP = (f, list) => {
+  const result = {}
+  const promises = []
+
+  for (const key in list) {
+    const p = f(list[key]).then(val => { result[key] = val })
+    promises.push(p)
+  }
+
+  return Promise.all(promises).then(() => result)
+}
+
+// mapP :: Functor f => (a -> Promise b) -> f a -> Promise f b
+const mapP = (f, list) => {
+  switch (Object.prototype.toString.call(list)) {
+    case '[object Object]':
```

Comment:
I like the idea of "mapping" async functions over object key-value pairs (which is why I even included [`mapObj` in `tinyfunk`](https://github.com/flintinatux/tinyfunk/blob/master/src.js#L151-L156)), but I'd rather avoid the ternary and not follow Ramda's decision to overload `map` by treating POJO's as functors.  I'd be more amenable to a separate function altogether named `mapObjP` that operates on just objects.

Reply:
:thinking: I like matching ramda's contract for better accessibility & integration with our existing code. We may need a tie breaker here.